### PR TITLE
[stdlib] Add OptionSet tests

### PR DESCRIPTION
**Pull request by lancep**
Test our public OptionSet API

This adds validation tests for the public API of OptionSet.

### CODE REVIEW

**Comment by lancep** [comment]
@swift-ci please smoke test

File: validation-test/stdlib/OptionSet.swift [hunk 1]
```diff
@@ -0,0 +1,776 @@
...
+  static let friday = Days(rawValue: 1 << 5)
+  static let saturday = Days(rawValue: 1 << 6)
+}
+
+// Test rawValue
+
+OptionSetTests.test("RawValue.SingleOption") {
+  let s = Days.friday
+  
+  assert(s.rawValue == 32)
```
Comment (moiseev) [inline validation-test/stdlib/OptionSet.swift]: Here and everywhere why `assert`?

Comment (moiseev) [reply validation-test/stdlib/OptionSet.swift]: `expectEqual` or `expectTrue`.

File: validation-test/stdlib/OptionSet.swift [hunk 2]
```diff
@@ -0,0 +1,776 @@
...
+
+OptionSetTests.test("UpdateWith.EmptySet") {
+  var s: Days = []
+
+  assert(s.contains(.monday) == false)
+  
+  let member1 = s.update(with: .monday)
+  
+  assert(s.contains(.monday))
+  expectEqual(member1, nil)
```
Comment (moiseev) [inline validation-test/stdlib/OptionSet.swift]: `expectNil`

File: validation-test/stdlib/OptionSet.swift [hunk 3]
```diff
@@ -0,0 +1,776 @@
...
+  
+  let member2 = s.update(with: .monday)
+  expectEqual(member2, .monday)
```
Comment (moiseev) [inline validation-test/stdlib/OptionSet.swift]: `expectOptionalEqual`

**Review by moiseev — CHANGES_REQUESTED** [review]
I don't remember whether we have the `SetAlgebra` conformance tests. But it would be nice to add a test for that too.

**Comment by lancep** [comment]
I have a separate bug for the SetAlgebra tests

**Review by milseman — APPROVED** [review]
LGTM

**Comment by lancep** [comment]
@swift-ci please smoke test

**Review by moiseev — APPROVED** [review]

**Pull request merged**